discovery_child_development/pipeline/openalex/00_openalex_metaflow.py

```diff
@@ -71,15 +74,26 @@ def api_generator(api_root: str, concept_ids: List[str]) -> list:
         all_pages: list of pages required to return all results
     """
     concepts_text = concept_ids
+
     page_one = f"{api_root}concepts.id:{concepts_text}"
     print(f"Running API query {page_one}")
+
     total_results = requests.get(page_one).json()["meta"]["count"]
     print(f"Total number of hits: {total_results}")
+
     number_of_pages = -(total_results // -200)  # ceiling division
-    all_pages = [
-        f"{api_root}concepts.id:{concepts_text}&per-page=200&cursor="
-        for _ in range(1, number_of_pages + 1)
-    ]
+    print(f"Total number of pages queried: {number_of_pages}")
```

Comment:
Finding it hard to get my head round why adding `&page={_}` helps get a random sample, maybe because I don't really understand paging with the OpenAlex API... Would you be able to add a comment here to explain the conditional?

Reply:
So according the OpenAlex api docs, if you are getting a random sample you have to use basic paging (so have to cycle through each page like that), rather than the cursor paging that is used (with the *) in the generate_api_calls. So you have to do it like that instead! See [here](https://docs.openalex.org/how-to-use-the-api/get-lists-of-entities/sample-entity-lists) and [here](https://docs.openalex.org/how-to-use-the-api/get-lists-of-entities/paging#basic-paging). This is also why you can only get a max of 10,000 results this way. 

Will add some info in!

Reply:
Amazing! That helps a lot